draft-ZIP-0226.rst

```diff
@@ -0,0 +1,303 @@
+
+**Asset Type Equality:** the following constraints must be added to ensure that the input and output note are of the
+same type:
+
+- The asset type, :math:`\mathsf{type_\mathbb{P}}`, for the note is witnessed once, as an auxiliary input.
+- The witnessed asset type, :math:`\mathsf{type_\mathbb{P}}`, is added to the old note commitment input.
+- The witnessed asset type, :math:`\mathsf{type_\mathbb{P}}`, is added to the new note commitment input.
+
+**Correct Value Commitment Type:** the following constraints must be added to ensure that the value commitment is computed using the witnessed type, as represented in the notes
+
+- The fixed-base multiplication constraints between the value and the value base point of the value commitment,:math:`\mathsf{cv}`, is replaced with a variable-base multiplication between the two
+- The witness to the value base-point is the auxiliary input :math:`\mathsf{type}_\mathbb{P}`.
+
+**Enforce Secure Type for Split Actions:** the following constraints must be added to prevent senders from malling with the asset type for the output note in the consequent actions of the split:
+
+- The Value Commitment Integrity should be changed
+    - Replace the input note value by a generic value, `v'`, as :math:`\mathsf{cv^net} = \mathsf{ValueCommit_rcv^OrchardType(v’ - v^new, type}_\mathbb{P})`
+- Add a boolean “split” variable as an auxiliary witness. This variable is to be activated `split = 1` if the Action in question is a split and `split = 0` if the Action is actually spending an input note:
+    - If `split = 1` then set `v' = 0` otherwise `v'=v^old` from the auxiliary input
+- The Merkle Path Validity should check the existance of the note commitment as usual (and not like with dummy notes):
+    - Check that (path, pos) is a valid Merkle path of depth :math:`\mathsf{MerkleDepth^Orchard}`, from :math:`\mathsf{cm^old}` to the anchor :math:`\mathsf{rt^Orchard}`.
+- The Nullifier Integrity will be changed to prevent the identification of notes
+    - Replace the :math:`\psi_{old}` value with a generic :math:`\psi'` as :math:`\mathsf{nf_old = DeriveNullifier_nk}(\rho^\mathsf{old}, \psi', \mathsf{cm^old})`
+    - if `split = 1` set :math:`\psi' = \mathsf{randomSample}`, otherwise set :math:`\psi' = \psi^{old}`
+
+**Enabling Backwards Compatibility with ZEC Notes:** the following constraints must be added to enable backwards compatibility with the Orchard ZEC notes.
+
+The old note commitment is computed using a “rolling-aggregate” sinsemilla commitment. This means that the commitment is computed by adding new chunks or windows to the accumulated value. This method will be used in order to maintain a single commitment instance for the old note commitment, that will be used both for Orchard ZEC notes and for ZSA notes. The original Orchard ZEC notes will be conserved and not actually be converted into ZSA notes, as we will always need to compute them.
```

Comment:
```suggestion
The old note commitment is computed using a “rolling-aggregate” Sinsemilla commitment. This means that the commitment is computed by adding new chunks or windows to the accumulated value. This method will be used in order to maintain a single commitment instance for the old note commitment, that will be used both for Orchard ZEC notes and for ZSA notes. The original Orchard ZEC notes will be conserved and not actually be converted into ZSA notes, as we will always need to compute them.
```

Reply:
This needs to be specified more precisely.

Reply:
Has been specified more formally, refer to PR#680 for the latest updates.